### ISSUE TITLE
ERROR:  integer out of range

### ISSUE THREAD

**Issue by Neo-Burner** [oar-team/oar#41]
Hi,
I'm working on migration from MySql to Postgres on Grid'5000 and during insertion of data I got this error : 

    ERROR:  integer out of range

It happens in accounting table because (I guess) window_start, window_stop and consumption are Integer and not BigInteger in PostGres DB

**Comment by capitn** [comment]
Hi,

Do you know the value of the integer that triggered this error message?

According to the postgresql documentation http://www.postgresql.org/docs/9.1/static/datatype-numeric.html , the maximum value for an integer is 2147483647. The corresponding date is so in 2038.

The mysql documentation https://dev.mysql.com/doc/refman/5.1/en/integer-types.html gives also 2147483647 for the max INT value.

**Comment by SalemHarrache** [comment]
In my G5K dump, only the `consumption` column exceeds
```
mysql> select max(window_start),max(window_stop),max(consumption) from accounting;
+-------------------+------------------+------------------+
| max(window_start) | max(window_stop) | max(consumption) |
+-------------------+------------------+------------------+
|        1454976000 |       1455062399 |       2316893560 |
+-------------------+------------------+------------------+
```

however, the column type is not an Integer, but a *unsigned* Integer:
```
mysql> describe accounting;
+--------------------+----------------------+------+-----+---------+-------+
| Field              | Type                 | Null | Key | Default | Extra |
+--------------------+----------------------+------+-----+---------+-------+
| window_start       | int(10) unsigned     | NO   | PRI | NULL    |       |
| window_stop        | int(10) unsigned     | NO   | PRI | NULL    |       |
| accounting_user    | varchar(255)         | NO   | PRI | NULL    |       |
| accounting_project | varchar(255)         | NO   | PRI | NULL    |       |
| queue_name         | varchar(100)         | NO   | PRI | NULL    |       |
| consumption_type   | enum('ASKED','USED') | NO   | PRI | NULL    |       |
| consumption        | int(10) unsigned     | NO   |     | NULL    |       |
+--------------------+----------------------+------+-----+---------+-------+
7 rows in set (0.00 sec)
```

```
oar=> \d+ accounting
                                                     Table "public.accounting"
       Column       |          Type          |                  Modifiers                  | Storage  | Stats target | Description 
--------------------+------------------------+---------------------------------------------+----------+--------------+-------------
 window_start       | integer                | not null                                    | plain    |              | 
 window_stop        | integer                | not null default 0                          | plain    |              | 
 accounting_user    | character varying(255) | not null default ''::character varying      | extended |              | 
 accounting_project | character varying(255) | not null default ''::character varying      | extended |              | 
 queue_name         | character varying(100) | not null default ''::character varying      | extended |              | 
 consumption_type   | character varying(5)   | not null default 'ASKED'::character varying | extended |              | 
 consumption        | integer                | not null default 0                          | plain    |              | 
Indexes:
    "accounting_pkey" PRIMARY KEY, btree (window_start, window_stop, accounting_user, accounting_project, queue_name, consumption_type)
    "accounting_project" btree (accounting_project)
    "accounting_queue" btree (queue_name)
    "accounting_type" btree (consumption_type)
    "accounting_user" btree (accounting_user)
Check constraints:
    "accounting_consumption_type_check" CHECK (consumption_type::text = ANY (ARRAY['ASKED'::character varying, 'USED'::character varying]::text[]))

```

**Comment by capitn** [comment]
Maybe we can change the consumption field into biginteger.
But the mysql documentation says that this is the same limit with signed and unsigned integer...

Does something like:

    alter table accounting alter column consumption type bigint ;

resolves the issue in the postgresql database?

**Comment by SalemHarrache** [comment]
The numeric type table of the mysql documentation is not easy to read regarding the unsigned, but if you look closely, you will see that there is a higher limitation for the unsigned integer (second row).

```
INT	4	-2147483648	2147483647
 	 	0			4294967295
```
Changing type to big integer resolves the issue for me.

**Comment by capitn** [comment]
Indeed I didn't read the doc correctly =)

@npf : do you think I put the change into the 2.5 branch right now for the 2.5.5 release?

**Comment by npf** [comment]
Yes. Freeze is not here yet..

**Comment by capitn** [comment]
Ok it will be fixed in 2.5.5 and later.

Thanks for reporting this bug.

**Comment by capitn** [comment]
Ok it will be fixed in 2.5.5 and later.

Thanks for reporting this bug.
